.github/workflows/build-container.yaml

```diff
@@ -109,19 +108,20 @@ jobs:
           pattern: tag*
       - id: test-image
         name: Test image against tests on main
+        env:
+          TAG: ${{ needs.build-image.outputs.test-tag }}
         run: |
-          tag=$(ls artifacts/)
-          path=$(ls artifacts/$tag/*)
+          path=$(ls artifacts/*/*)
           docker load --input "$path"
           tests=$(mktemp)
           curl -o $tests https://raw.githubusercontent.com/hubverse-org/hub-dash-site-builder/refs/heads/main/tests/run.sh
-          bash ${tests} "ghcr.io/hubverse-org/hub-dash-site-builder:${tag}" reichlab/flusight-dashboard
+          bash ${tests} "ghcr.io/hubverse-org/hub-dash-site-builder:${TAG}" reichlab/flusight-dashboard
         shell: bash
   publish:
     needs: [build-image]
     name: "Publish Image"
     runs-on: ubuntu-latest
-    if: ${{ github.event_name != 'pull_request' }}
+    if: ${{ fromJSON(env.PUBLISH) }}
```

Comment:
Only running publish image if we have the publishing conditions (workflow dispatch from main or a tagged release)